src/components/v5/common/CompletedAction/partials/PaymentBuilder/PaymentBuilder.tsx

```diff
@@ -142,6 +148,87 @@ const PaymentBuilder = ({ action }: PaymentBuilderProps) => {
     },
   ];
 
+  if (expenditure.type === ExpenditureType.Staged) {
+    return (
+      <>
+        <div className="flex w-full items-center justify-between gap-2">
+          <ActionTitle>{customTitle}</ActionTitle>
+          <MeatBallMenu
+            contentWrapperClassName={clsx('z-[65] sm:min-w-[11.25rem]', {
+              '!left-6 right-6': isMobile,
+            })}
+            dropdownPlacementProps={{
+              top: 12,
+            }}
+            items={expenditureMeatballOptions}
+          />
+        </div>
+        <ActionSubtitle>
+          {formatText(
+            { id: 'action.title' },
+            {
+              actionType: ExtendedColonyActionType.StagedPayment,
+              initiator: initiatorUser ? (
+                <UserInfoPopover
+                  walletAddress={initiatorUser.walletAddress}
+                  user={initiatorUser}
+                  withVerifiedBadge={false}
+                >
+                  {initiatorUser.profile?.displayName}
+                </UserInfoPopover>
+              ) : null,
+              recipient: 'qwe',
+            },
+          )}
+        </ActionSubtitle>
+        <ActionDataGrid>
+          <ActionTypeRow actionType={ExtendedColonyActionType.StagedPayment} />
+
+          {selectedTeam?.metadata && (
+            <TeamFromRow
+              teamMetadata={selectedTeam.metadata}
+              actionType={action.type}
+            />
+          )}
+
+          <ActionData
+            rowLabel={formatText({ id: 'actionSidebar.recipient' })}
+            rowContent={
+              <UserPopover
+                walletAddress={slots?.[0].recipientAddress || ADDRESS_ZERO}
+                size={20}
+              />
+            }
+            RowIcon={UserFocus}
+            tooltipContent={formatText({
+              id: 'actionSidebar.tooltip.simplePayment.recipient',
+            })}
+          />
+
+          <DecisionMethodRow
+            decisionMethod={isStaked ? DecisionMethod.Staking : undefined}
+            isMotion={action.isMotion || false}
+          />
+
+          {action.motionData?.motionDomain.metadata && (
+            <CreatedInRow
+              motionDomainMetadata={action.motionData.motionDomain.metadata}
+            />
+          )}
+        </ActionDataGrid>
+        {action.annotation?.message && (
+          <DescriptionRow description={action.annotation.message} />
+        )}
+        <StagedPaymentTable
+          stages={expenditure.metadata?.stages || []}
+          slots={slots}
+          isLoading={!expenditure.metadata?.stages?.length}
+          isReleaseStep={expenditureStep === ExpenditureStep.Release}
+        />
+      </>
+    );
+  }
```

Comment:
There's a fair bit of duplication between the two snippets which makes me wonder if we can reduce it somehow? Also it might be worth splitting this into separate components depending on expenditure type

Reply:
@jakubcolony I've refactored whole component